### PR TITLE
docs: Improved the C# documentation across multiple pages

### PR DESCRIPTION
**Pull request by jamesmcroft**
## Proposed changes

Having been using Appium & C# as a base for my Legerity UI testing framework, I noticed a lack of documented samples and links for the .NET implementations of the Appium APIs.

As such, I've gone through as many of the APIs as I am possibly aware of/have used myself, and have provided a sample following the patterns of the existing language samples and including a link to the original documented source code where possible.

I hope that by adding more samples in, this will help developers looking to adopt Appium with .NET in their application UI testing as I do today.

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: commands-yml/commands/device/app/install-app.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ client_docs:
   ruby: "https://www.rubydoc.info/github/appium/ruby_lib_core/Appium/Core/Device#install_app-instance_method"
   ruby_core: "https://www.rubydoc.info/github/appium/ruby_lib_core/Appium/Core/Device#install_app-instance_method"
   php: "https://github.com/appium/php-client/" # TODO PHP documentation link
-  csharp: "https://github.com/appium/appium-dotnet-driver/" # TODO Dotnet documentation link
+  csharp: "https://github.com/appium/appium-dotnet-driver/blob/master/src/Appium.Net/Appium/AppiumDriver.cs#L227"
```
Comment (mykola-mokhnach) [inline commands-yml/commands/device/app/install-app.yml:43]: Please do not use line numbers in links. These might easily get outdated if the file is changed

**Comment by mykola-mokhnach** [comment]
Thanks a lot @jamesmcroft 

We always appreciate documentation updates

File: commands-yml/commands/device/app/install-app.yml (hunk 1 above)
Comment (jamesmcroft) [reply commands-yml/commands/device/app/install-app.yml:43]: Good point, I've done this throughout so will remove them 👍

Comment (jamesmcroft) [reply commands-yml/commands/device/app/install-app.yml]: Should be sorted now, thanks for the feedback 😄

**Review by mykola-mokhnach — APPROVED** [review]

**Review by KazuCocoa — APPROVED** [review]
thanks so much 🎉

**Comment by jamesmcroft** [comment]
@KazuCocoa more than welcome 😄

**Pull request merged**